File: frontend/lib/types/job.ts
export interface JobItem {
  jobId: number;
  jobTitle: string;
  jobStatus: string;
  department: string;
  createdAt: string;
  applicantCount: number;
}

export interface InsertJobPayload {
  jobTitle: string;
  location: string;
  jobType: string;
  jobStatus: string;
  jobDescription: string;
  qualifications: string[];
  techSkills: string[];
  department: string;
}

export interface JobDetailsResponse {
  jobId: number;
  jobTitle: string;
  department: string;
  location: string;
  jobType: string;
  jobStatus: string;
  jobDescription: string;
  qualifications: string[];
  techSkills: string[];
  createdAt: string;
}

export interface InsertJobResponse {
  message: string;
  jobDetails: InsertJobPayload;
}

export interface GetAllJobsResponse {
  data: JobItem[];
  totalCount: number;
  pageNumber: number;
  pageSize: number;
  totalPage: number;
}

export interface EditJobPayload {
  jobTitle: string;
  location: string;
  jobType: string;
  jobStatus: string;
  jobDescription: string;
  qualifications: string[];
  techSkills: string[];
  department: string;
}

File: backend/ai-recruitment/Features/JobPosts/Model/Job.cs
using ai_recruitment.Features.Candidates.model;
using System.ComponentModel.DataAnnotations;

namespace ai_recruitment.Features.JobPosts.Model
{
    public class Job
    {
        [Key]
        public int JobId { get; set; }
        public required string JobTitle { get; set; }
        public required string Location { get; set; }
        public required string JobType { get; set; }
        public required string JobStatus { get; set; }
        public required string JobDescription { get; set; }
        public required string Department { get; set; }
        public required string[] Qualifications { get; set; }
        public required string[] TechSkills { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigation property: one job -> many applications
        public ICollection<Candidate> Candidates { get; set; } = new List<Candidate>();
    }
}
